### PR TITLE
Do not add bad connections back to pool (fix for #452)

### PR DESCRIPTION
**Pull request by elliotf**
See #452.

This is a hacky workaround proof-of-concept conversation starter that
fixes the issue, but should be done in a cross-dialect way, and it
should also include tests.

### CODE REVIEW

**Comment by tgriesser** [comment]
@elliotf is this one still good to be merged?

File: lib/pool.js [hunk 1]
```diff
@@ -32,6 +40,11 @@ Pool.prototype.defaults = function() {
     create: function(callback) {
       pool.client.acquireRawConnection()
         .tap(function(connection) {
+          if ('function' === typeof connection.on) {
+            var errorHandler = connectionErrorHandler.bind(null, pool, connection);
+            connection.on('error', errorHandler);
+            connection.on('end',   errorHandler);
```
Comment (atiertant) [inline lib/pool.js]: 'end' event could be emit without any error

File: lib/pool.js [hunk 2]
```diff
@@ -23,6 +23,14 @@ Pool.prototype.initialize = function() {
   return new GenericPool(_.defaults(this.config, _.result(this, 'defaults')));
 };
 
+function connectionErrorHandler(pool, connection, err) {
+  if (connection && err && err.fatal) {
```
Comment (atiertant) [inline lib/pool.js]: does not handle 'end' event if there is no err

**Comment by atiertant** [comment]
an other problem is "what's happening to query using this bad connection ?" : Promise is never resolved or rejected ...

**Comment by elhigu** [comment]
Hi, I started going through old pull requests and get them merged if possible or close otherwise if they are not going to be finished.

@elliotf do you think this is still relevant? If so, conflicts needs to be fixed also there was already couple of code comments. If PR is fixed I'll give it a review too.

**Comment by elhigu** [comment]
Closing this for now since this hasn't been active, lets open this again if work continues... @wubzz does your changes to pooling do anything related to this issue?

**Pull request closed without merging**